pkg/virtual/framework/transforming/transformers.go

```diff
@@ -0,0 +1,229 @@
+/*
+Copyright 2021 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package transforming
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/klog/v2"
+)
+
+type UpdateSelectorsFunc func(ctx context.Context, labelSelector labels.Selector, fieldSelector fields.Selector) (labels.Selector, fields.Selector, error)
+
+func TransformsSelectors(transformerName string, updateFunc UpdateSelectorsFunc) Transformer {
+
+	updateSelectorString := func(ctx context.Context, labelSelectorStr string, fieldSelectorStr string) (newLabelSelectorStr string, newFieldSelectorStr string, err error) {
+		labelSelector, err := labels.Parse(labelSelectorStr)
+		if err != nil {
+			return "", "", err
+		}
+		fieldSelector, err := fields.ParseSelector(fieldSelectorStr)
+		if err != nil {
+			return "", "", err
+		}
+		newLabelSelector, newFieldSelector, err := updateFunc(ctx, labelSelector, fieldSelector)
+		if err != nil {
+			return "", "", err
+		}
+		newLabelSelectorStr = newLabelSelector.String()
+		newFieldSelectorStr = newFieldSelector.String()
+		return
+	}
+
+	return Transformer{
+		Name: transformerName,
```

Comment:
named objects are often suspicious. Why does a transformer need a name?

Reply:
Mainly for logging